### PR TITLE
Add Fuzz workflow: daily Monte Carlo fuzzing of the yolo CLI

### PR DESCRIPTION
**Pull request by glenn-jocher**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

**WARNING ⚠️** this PR is very large, summary may not cover all changes.

### 🌟 Summary
Adds a new scheduled YOLO CLI fuzzing system 🧪 plus CI reliability tweaks and broad dataset documentation refreshes 📚

### 📊 Key Changes
- Added `.github/scripts/fuzz.py`, a Monte Carlo fuzzing tool for the `yolo` CLI with `fuzz`, `repro`, and `report` subcommands.
- Introduced a new daily **Fuzz** GitHub Actions workflow across multiple OS/personality shards: train, export, predict/val, and chaos 🖥️.
- Automatically classifies fuzz outcomes such as passes, expected validation errors, flakes, hangs, crashes, and bug candidates.
- Confirms reproducible findings, deduplicates them by stable signatures, and can auto-file capped GitHub issues with the `fuzz` label 🐛.
- Added manual repro support via workflow input to rerun exact fuzz-discovered commands.
- Made Codecov uploads non-blocking in CI to reduce failures caused by external coverage service issues.
- Updated Docker workflow commands to use `docker run --rm` and precache test assets before container tests 🐳.
- Refreshed many dataset docs, especially classification datasets and selected detection datasets, with clearer descriptions, counts, splits, usage examples, citations, FAQs, and Ultralytics Platform links.

### 🎯 Purpose & Impact
- Improves YOLO CLI robustness by continuously testing randomized and edge-case command combinations beyond the normal test matrix 🚀.
- Helps developers find real crashes, hangs, and validation gaps earlier, with reproducible commands and environment details attached.
- Reduces CI noise from flaky third-party Codecov failures while preserving coverage uploads when available.
- Keeps Docker CI cleaner and more reliable by removing containers after runs and ensuring assets are cached before tests.
- Makes dataset documentation more accurate and beginner-friendly, helping users choose datasets, understand splits, and train YOLO26 classification or detection models with less confusion.
- Overall, this PR strengthens automated quality assurance while improving documentation clarity for both developers and everyday YOLO users ✅.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms your Infinite CI fuzzing workflow contribution has been received, and an Ultralytics engineer will assist with review. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: .github/scripts/fuzz.py [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+    log.info(
+        f"[fuzz] done: {n} trials {counters}, {len(findings)} confirmed unique findings, infra_failed={infra_failed}"
+    )
+
+
+def cmd_repro(args):
+    """Replay one exact command several times through the classifier and print the verdict."""
+    uni = load_universe()
+    log = uni["logger"]
+    argv = args.command.split()
```
Comment (UltralyticsAssistant) [inline .github/scripts/fuzz.py:499]: 💡 **MEDIUM**: `repro` is documented/workflow-described as replaying an exact command from a fuzz issue, and the generated issue body includes commands prefixed with `yolo `. Passing that string here leaves `yolo` in `argv`, so `run_trial()` invokes `python -m ultralytics.cfg.__init__ yolo ...`; the CLI parser treats `yolo` as an unknown argument and the repro path reports a validation error instead of replaying the finding. Strip an optional leading `yolo` before building the trial.

**Suggested change:**
```suggestion
    argv = args.command.split()[1:] if args.command.split()[:1] == ["yolo"] else args.command.split()
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

One functional issue found: the repro subcommand cannot correctly replay the exact commands emitted in fuzz issues because it does not handle the leading `yolo` token. The rest of the fuzz/report flow looks coherent from the diff.

💬 Posted 1 inline comment

File: .github/scripts/fuzz.py (hunk 1 above)
Comment (glenn-jocher) [reply .github/scripts/fuzz.py:499]: Fixed in 73fac80a99 — `repro` now strips an optional leading `yolo` token, and additionally remaps runner-local absolute `model=`/`source=` paths from issue commands to this machine's cached copies so pasted commands replay portably.

**Comment by UltralyticsAssistant** [comment]
🚀 Merged — thank you @glenn-jocher, with valuable contributions from @JESUSROYETH, @Y-T-G, @sokrisba, @dorianboille, @raimbekovm, and @dependabot[bot]!

“Quality is not an act, it is a habit.” — Aristotle

This PR is a strong step toward making that habit part of every YOLO workflow: daily CLI fuzzing, reproducible failure reporting, stronger CI reliability, cleaner Docker runs, and refreshed dataset documentation all help make Ultralytics more robust, approachable, and dependable for the community. The new fuzzing system especially expands our ability to catch edge cases early, while the documentation updates make it easier for users to train and deploy YOLO26 models with confidence.

Appreciate the thoughtful engineering and documentation effort behind this one 🙌
